test/integration-new/snat/snat_test.go

```diff
@@ -0,0 +1,153 @@
+	It("Pod in private subnet should have Internet access with External SNAT disabled", func() {
+		By("Setting External SNAT to true")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_EXTERNALSNAT": "false",
+		})
+
+		By("Checking External Domain Connectivity")
+		ValidateExternalDomainConnectivity(EXTERNAL_DOMAIN)
+	})
+
+	It("Pod in private subnet should have Internet access with External SNAT enabled", func() {
+		By("Setting External SNAT to true")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_EXTERNALSNAT": "true",
+		})
+
+		By("Checking External Domain Connectivity")
+		ValidateExternalDomainConnectivity(EXTERNAL_DOMAIN)
+	})
+
+	It("Verify SNAT IP table rule by changing AWS_VPC_K8S_CNI_RANDOMIZESNAT", func() {
+		By("Check whether SNAT IP table has random-fully with AWS_VPC_K8S_CNI_RANDOMIZESNAT set to default value of prng")
+		ValidateIPTableRules("prng")
+
+		By("Setting AWS_VPC_K8S_CNI_RANDOMIZESNAT to none")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_RANDOMIZESNAT": "none",
+		})
+
+		By("Check where SNAT IP table rule is updated and it doesn't contain random port allocation")
+		ValidateIPTableRules("none")
+	})
+
+	It("Verify External Domain Connectivity by modifying AWS_VPC_K8S_CNI_EXCLUDE_SNAT_CIDRS", func() {
+		By("Getting CIDR for primary node's private subnet")
+		out, err := f.CloudServices.EC2().DescribeSubnet(privateSubnetId)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(out.Subnets)).To(BeNumerically(">", 0))
+
+		cidrBlock := out.Subnets[0].CidrBlock
+		By("Updating AWS_VPC_K8S_CNI_EXCLUDE_SNAT_CIDRS with private subnet CIDR")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_EXCLUDE_SNAT_CIDRS": aws.StringValue(cidrBlock),
+		})
+
+		By("Check External domain connectivity from this private subnet CIDR block")
+		ValidateExternalDomainConnectivity(EXTERNAL_DOMAIN)
+	})
+
+	AfterEach(func() {
+		By("Reverting aws-node env variables to default values")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_EXTERNALSNAT":  "false",
+			"AWS_VPC_K8S_CNI_RANDOMIZESNAT": "prng",
+		})
+		k8sUtils.RemoveVarFromDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]struct{}{
+			"AWS_VPC_K8S_CNI_EXCLUDE_SNAT_CIDRS": {},
+		})
+	})
+})
+
+func ValidateExternalDomainConnectivity(url string) {
+	testerArgs := []string{
+		"-testExternalDomainConnectivity=true",
+		fmt.Sprintf("-url=%s", url),
+	}
+
+	testContainer := manifest.NewTestHelperContainer().
```

Comment:
This could be achieved by something simpler like `curl --fail URL` by checking for non zero exit code instead of creating a new utility in agent IMO